Gordon360/Services/HousingService.cs

```diff
@@ -558,26 +546,40 @@ public bool ChangeApplicationEditor(string editorId, int apartAppId, string newE
                 return false;
             }
 
-            ApartmentAppEditorViewModel editorModel = editorResult.ElementAt(0);
-            string storedEditorId = editorModel.EditorID;
+            string storedEditorID = editorResult.FirstOrDefault();
 
-            if (editorId != storedEditorId)
+            if (userID != storedEditorID)
             {
                 // Return false if the current user does not match this application's editor stored in the database
                 return false;
             }
             // Only perform the update if the ID of the current user matched the 'EditorID' ID stored in the database for the requested application
 
-            appIdParam = new SqlParameter("@APPLICATION_ID", apartAppId);
-            editorParam = new SqlParameter("@EDITOR_ID", editorId);
-            timeParam = new SqlParameter("@NOW", now);
-            newEditorParam = new SqlParameter("@NEW_EDITOR_ID", newEditorId);
+            IEnumerable<ApartmentApplicationViewModel> result = null;
+
+            DateTime now = System.DateTime.Now;
 
-            result = RawSqlQuery<ApartmentAppSaveResultViewModel>.query("UPDATE_AA_APPLICATION_EDITOR @APPLICATION_ID, @EDITOR_ID, @NOW, @NEW_EDITOR_ID", appIdParam, editorParam, timeParam, newEditorParam); //run stored procedure
-            if (result == null)
+            appIDParam = new SqlParameter("@APPLICATION_ID", applicationID);
+            SqlParameter timeParam = new SqlParameter("@NOW", now);
+            if (newEditorID != storedEditorID)
             {
-                throw new ResourceNotFoundException() { ExceptionMessage = "The application could not be updated." };
+                SqlParameter editorParam = new SqlParameter("@EDITOR_ID", userID);
+                SqlParameter newEditorParam = new SqlParameter("@NEW_EDITOR_ID", newEditorID);
+                result = RawSqlQuery<ApartmentApplicationViewModel>.query("UPDATE_AA_APPLICATION_EDITOR @APPLICATION_ID, @EDITOR_ID, @NOW, @NEW_EDITOR_ID", appIDParam, editorParam, timeParam, newEditorParam); //run stored procedure
+                if (result == null)
+                {
+                    throw new ResourceCreationException() { ExceptionMessage = "The application could not be updated." };
+                }
             }
+            else
```

Comment:
Why do  we only update the dateModified if the editor is not changed?

Reply:
Because the dateModified is automatically updated when you update the editor. 
i.e. no need to update the dateModified again, since it was already updated just now

Reply:
Oh, right! We update date modified in the stored procedure- is that it? That makes things confusing, doesn't it

Reply:
We could separate it into two separate procedures, 

but actually the thing we should do instead is try to make one stored procedure that will do the entire update all in one stored procedure, so that the update action becomes an ACID transaction and also there are other benefits. It is very easily possible and I also think it is possible or even better to do some of the comparison stuff in there as well, because I think you can check for all the add/update/remove stuff far more efficiently with SQL commands rather than manually doing it in the for loops like we have above for the applicants and apartmentChoices. 

The SQL commands are specifically designed for manipulating the data in the tables in all of those ways, so it makes sense to perform those actions natively in an SQL stored procedure rather than having to code-out the comparison algorithms by hand as we have done here. (It would be a more concise, efficient, elegant solution)

This is something I would like to look into later if we get the MVP features fully functional

Reply:
It seems like this would make testing very tedious, though, as we would have to create a whole application to create a test applicant or hall choice. Like you said, though, something to consider later

Reply:
That is not true, because we would still keep around the stored procedures for modifying just those specific parts. It doesn't have to be one or the other, it makes sense to have both so that you can either just throw a full application at it, or selectively say "only update the halls, leave the rest alone". both are important features that should be there
We should prioritize the functionality and reliability of the feature over the convenience or tediousness of writing a unit test for it. 
We will need those tests anyway, 
we need to be able to unit test a full application, because that it the most common real-world scenario and needs to work reliably

Reply:
Referenced or described [here](https://github.com/gordon-cs/gordon-360-api/pull/497#discussion_r599016060)